### ISSUE TITLE
Routing error with Attribute error if coupling_map is None

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit#7127]
### Information

- **Qiskit Terra version**: main
- **Python version**:
- **Operating system**:

### What is the current behavior?

All the routing passes `StochasticSwap`, `SabreSwap` (after #7111 ), `LookaheadSwap`, and `BasicSwap` all enable you to initialize the pass with `coupling_map=None` but when you call run() on the pass they all fail with the same attribute error:

```
AttributeError: 'NoneType' object has no attribute 'physical_qubits'
```

This doesn't come up in the preset pass managers because they skip the execution of routing passes if the coupling_map is not set.

### Steps to reproduce the problem

```python
from qiskit.circuit.random import random_circuit
from qiskit.transpiler.passes import StochasticSwap, SabreSwap
stoch = StochasticSwap(coupling_map=None)
qc = random_circuit(5, 5)
stoch(qc)
```

### What is the expected behavior?

Either the pass fails with a `TranspilerError` describing that routing can't be run without a coupling map, or we simply return the dag because without a coupling map there is no routing to be done so it should be a no-op.

### Suggested solutions

Fix the logic in the routing passes to handle `coupling_map=None`

**Comment by mcbennet** [comment]
Hi, 

I would like to attempt to resolve this issue.

Best,

**Comment by jakelishman** [comment]
@mcbennet: in case you didn't see, the issue was assigned to you (I don't think you get notified).

**Comment by mtreinish** [comment]
Closed by #7141
